.chloggen/ottl-grammar-custom-errors-visitor.yaml

```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: pkg/ottl
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Refactor grammar's validation to use the AST visitor and to combine all errors."
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [35728]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext: Parsing invalid statements and conditions now combines all errors instead of returning the first one found.
```

Comment:
```suggestion
note: "Parsing invalid statements and conditions now prints all errors instead of just the first one found."

# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
issues: [35728]

# (Optional) One or more lines of additional information to render under the primary note.
# These lines will be padded with 2 spaces and then inserted directly into the document.
# Use pipe (|) for multiline entries.
subtext:
```
Minor, but I think as cool as this refactoring is, we should probably trim down the changelog entry since most users will be focused on the end result.

Reply:
It makes sense, thanks for reviewing @evan-bradley!